keps/sig-apps/2255-pod-cost/README.md

```diff
@@ -55,101 +48,133 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+One risk is that users perceive the feature as a guarantee to delete order. Documentation
+should stress the fact that this is best effort.
 
 ## Design Details
 
+The pod-cost range will be from [-MaxInt, MaxInt]. The default value is 0.
+Invalid values (like setting the annotation to string) will be converted to
+the default value.
+
+When scaling down a ReplicaSet, controller-manager will prioritize deleting
+pods with lower `pod-cost`. Specifically, the pod-cost will be evaluated after step 3
+and before step 4 as they are currently defined in
+[ActivePodsWithRanks](https://github.com/kubernetes/kubernetes/blob/cac933934b1301665e6e51a81c66c483f4e16c49/pkg/controller/controller_utils.go#L784-L809),
+which means the followig criteria is applied when comparing two pods regardless of their pod-cost:
+- if one is assigned a node and the other is not, then the unassigned pod is deleted first.
+- if the two pods are in different phases, then the pod in pending/unknown status is deleted first.
+- if the two pods have different readiness status, then the not ready pod is deleted first
+
+
+If none of the pods set the pod-cost annotation or all of them have the same value, then the 
+scale down behavior is not changed compared to now.
 
 ### Test Plan
 
 * Units test in kube-controller-manager package to test a variety of scenarios.
-* New E2E Tests to validate that replicas get deleted as expected e.g:
- * Replicas with lower pod-cost before replicas with higher pod-cost
- * Replicas with no pod-cost annotation set before replicas with low priority
+* Integration tests to validate that:
+ * Replicas with lower pod-cost are deleted before replicas with higher pod-cost
+ * No behavior change when pod-cost is not set or all pods have the same pod-cost
 
 ### Graduation Criteria
 
 #### Alpha -> Beta Graduation
 * Implemented feedback from alpha testers
-* Thorough E2E and unit testing in place
 
 #### Beta -> GA Graduation
-* Significant number of end-users are using the feature
 * We're confident that no further API changes will be needed to achieve the goals of the KEP
 * All known functional bugs have been fixed
 
 ### Upgrade / Downgrade Strategy
 
-When upgrading no changes are needed to maintain existing behaviour as all of this behaviour is fully optional and disabled by default. To activate this feature either a user has to make an annotation to a pod in a Deployment by hand or the application annotates a pod in a Deployment through the API.
-
-When downgrading, there is no need to changing anything, as this is just a pod annotation, which is uncritical.
+N/A
```

Comment:
We should discuss what happens after cluster upgrades / downgrades, such as how the controller handles it on downgrade.

Reply:
There is no strategy here really, I added some text to that effect.